app/controllers/usermatching/HomeController.scala

```diff
@@ -81,6 +76,39 @@ class HomeController @Inject()(val auditingService: AuditingService,
         }
     }
 
+  private def handleNoSubscriptionFound(utr: String, timestamp: String, nino: String)
+                                       (implicit hc: HeaderCarrier, request: Request[AnyContent]) = {
+    getEligibilityStatusService.getEligibilityStatus(utr) flatMap {
+      // Check eligibility (this is complete, and gives us the control list response including pre-pop information)
+      case Right(EligibilityStatus(true, _, prepopMaybe)) =>
+        prePopulate(utr, prepopMaybe) map { _ =>
+          goToSignUp(timestamp)
+            .addingToSession(UTR -> utr)
+            .addingToSession(NINO -> nino)
+        }
+      case Right(EligibilityStatus(false, _, _)) =>
+        Redirect(eligibilityRoutes.NotEligibleForIncomeTaxController.show())
+      case Left(_) => throw new InternalServerException(s"[HomeController] [index] Could not retrieve eligibility status")
+    }
+  }
+
+  private def prePopulate(utr: String, prepopMaybe: Option[PrePopData])
+                         (implicit hc: HeaderCarrier, request: Request[AnyContent]) = {
+    if (isEnabled(PrePopulate)) {
+      prepopMaybe match {
+        case None => Future.successful(Unit)
+        // Check presence of user information from ITSA data store by requesting an optional boolean of pre-pop flag.
+        case Some(prePop) => subscriptionDetailsService.fetchPrePopFlag(utr) map {
+          case None =>
+            populateSubscription(utr, prePop)
```

Comment:
```suggestion
            populateSubscription(reference, prePop)
```
shouldn't be using utr to do this.
use the following to get the reference to save data against
```
withReference { reference =>
   ...
}
```

Reply:
Done.